### ISSUE TITLE
DockerTools only supports registry API v1, hub.docker.com hasn't supported v1 in over a year

### ISSUE THREAD

**Issue by grahamc** [NixOS/nixpkgs#21651]
## Issue description

DockerTools only supports registry API v1, hub.docker.com hasn't supported v1 in over a year:

https://blog.docker.com/2015/10/docker-hub-deprecation-1-5/

### Steps to reproduce

1. `default.nix`:

```nix
{ dockerTools }:
dockerTools.pullImage {
  imageName = "alpine";
  imageTag = "latest";
  imageId = null;
  sha256 = "1bhw5hkz6chrnrih0ymjbmn69hyfriza2lr550xyvpdrnbzr4gk2";

  indexUrl = "https://index.docker.io";
  registryVersion = "v1";
}
```

2. `nix-build` 

```
these derivations will be built:
  /nix/store/1mg5sascdlv281kz45z6rnbjgw2d3fa0-alpine-latest.drv
  /nix/store/6sii94lc1k2kr4d9gdsyd4arkb7klgb9-alpine-latest.tar.gz.drv
building path(s) ‘/nix/store/6rjvi0727zinb8mi5jjs761z5cz3iabh-alpine-latest’
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100     2    0     2    0     0      1      0 --:--:--  0:00:01 --:--:--     7
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0
curl: (22) The requested URL returned error: 404 NOT FOUND
builder for ‘/nix/store/1mg5sascdlv281kz45z6rnbjgw2d3fa0-alpine-latest.drv’ failed with exit code 22
cannot build derivation ‘/nix/store/6sii94lc1k2kr4d9gdsyd4arkb7klgb9-alpine-latest.tar.gz.drv’: 1 dependencies couldn't be built
error: build of ‘/nix/store/6sii94lc1k2kr4d9gdsyd4arkb7klgb9-alpine-latest.tar.gz.drv’ failed
```

## Technical details

* System: all
* Nix version: all
* Nixpkgs version: "17.03pre98073.1fa75a5" (but all up to this point)

**Comment by LnL7** [comment]
I think it's been deprecated for a (over)year, but still worked until recently or a few months ago.

**Comment by grahamc** [comment]
I don't think that is possible: "On December 7, 2015, pulls via clients 1.5
and earlier will be disabled."

**Comment by LnL7** [comment]
Oh, I'm surprised people where still using it afterwards then.

**Comment by fpletz** [comment]
cc @lethalman

**Comment by moretea** [comment]
I recently tried to download an image from the docker hub via nix, that failed. I think we should remove all broken tools. Note that the generation of `.tar.gz` files that can be loaded into docker with `docker load` still works and is quite useful.

**Comment by fpletz** [comment]
cc @lo1tuma

**Comment by lo1tuma** [comment]
Wasn’t this fixed by #26131?

**Comment by globin** [comment]
Fixed on master
